v2/yarpchttpfx/yarpchttpfx.go

```diff
@@ -155,7 +159,7 @@ type ClientParams struct {
 	fx.In
 
 	Config          OutboundsConfig
-	DialerProvider  yarpc.DialerProvider
+	Dialer          *yarpchttp.Dialer
```

Comment:
@kriskowal doesn't this prevent us from having dialer specific options on a per outbound service basis? I imagine that there are some fine tuned options that some users may want to change depending on the service. 

eg A service speaking gRPC to two services, one with TLS and the other without. We'd need two `yarpchttp.Dialer`s.

Reply:
You’re right that we probably need support for named dialers. We should keep this as a reasonable default.

When we do add named dialers, we should guard to make sure the `yarpc.Dialer` we get is concretely a `*yarpchttp.Dialer` during startup.